File: array/0167_two_sum_ii_input_array_is_sorted/0167_two_sum_ii_input_array_is_sorted.go
package main

import (
	"fmt"
)

func twoSum(numbers []int, target int) []int {

	start, end := 0, len(numbers)-1
	for start != end {
		sum := numbers[start] + numbers[end]
		if sum > target {
			end -= 1
		} else if sum < target {
			start += 1
		} else {
			break
		}
	}
	return []int{start + 1, end + 1}
}

func main() {
	numbers := []int{2, 7, 11, 15}
	target := 9
	fmt.Println(twoSum(numbers, target))
}

File: array/0167_two_sum_ii_input_array_is_sorted/0167_two_sum_ii_input_array_is_sorted.py
# -*- coding: utf-8 -*-


class Solution(object):
    def twoSum(self, numbers, target):
        """
        :type numbers: List[int]
        :type target: int
        :rtype: List[int]
        """
        d = {}
        for idx, num in enumerate(numbers):
            if target - num in d:
                return [d[target-num] + 1 , idx + 1 ]
            d[num] = idx



numbers = [2,7,11,15]
target = 9
print(Solution().twoSum(numbers, 9))